src/nvim/screen.c

```diff
@@ -2903,24 +2869,32 @@ win_line (
         /* Display the absolute or relative line number. After the
          * first fill with blanks when the 'n' flag isn't in 'cpo' */
         if ((wp->w_p_nu || wp->w_p_rnu)
-            && (row == startrow
-                + filler_lines
+            && (row == startrow + filler_lines
                 || vim_strchr(p_cpo, CPO_NUMCOL) == NULL)) {
-          /* Draw the line number (empty space after wrapping). */
-          if (row == startrow
-              + filler_lines
-              ) {
+          // If 'signcolumn' is set to 'number' and a sign is present
+          // in 'lnum', then display the sign instead of the line
+          // number.
+          if (is_sign_number
+              && buf_findsign_id(wp->w_buffer, lnum, (char_u *)"*") != 0) {
+            int count = win_signcol_count(wp);
+            get_sign_display_info(true, wp, lnum, row,
+                startrow, filler_lines, filler_todo, count,
+                &c_extra, &c_final, extra, sizeof(extra),
+                &p_extra, &n_extra,
+                &char_attr, &draw_state, &sign_idx);
+          } else if (row == startrow + filler_lines) {
```

Comment:
> I cannot fix the last errors...

I think the problem is that you are making this place an `else if` statement. You must include up to line 2940 in the `else` statement.

Reply:
Thanks.